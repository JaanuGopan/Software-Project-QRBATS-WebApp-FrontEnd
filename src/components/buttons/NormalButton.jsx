import React from 'react';
import './normal-button.css';

const NormalButton = ({ title, icon, handleClick }) => {
  return (
    <div className="row my-1 align-items-center normal-button-container" onClick={handleClick}>
      <div className="col">
        <div className="row py-2">
          <div className="col-auto pe-0">{icon}</div>
          <div className="col-auto fs-6">{title}</div>
        </div>
      </div>
    </div>
  );
};

export default NormalButton;
