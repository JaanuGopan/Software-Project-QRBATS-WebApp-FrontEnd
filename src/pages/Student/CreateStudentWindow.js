import React, { useState } from "react";
import "./Student.css";
import axios from "axios";
import Designer from "../../assets/Images/Designer.jpeg";
import { useNavigate } from "react-router-dom";
import InputField from "../../components/textfields/InputBox/InputField";
import { IoMdCloseCircleOutline } from "react-icons/io";

const CreateStudentWindow = ({ handlecloseCreateStudentWindow }) => {
  const [firstname, setFirstName] = useState("");
  const [lastname, setLastName] = useState("");
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [userName, setUserName] = useState("");
  const [userRole, setUserRole] = useState("");
  const [departmentId, setDepartmentId] = useState("");
  const navigate = useNavigate();

  const deparmentList = ["DEIE", "DCOM", "DMME", "DCEE", "DMENA"];
  const userRoleList = ["ADMIN", "LECTURER", "STAFF"];

  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      const response = await axios.post(
        "http://localhost:8080/api/v1/auth/signin",
        {
          firstName: firstname,
          lastName: lastname,
          email: email,
          password: password,
          userName: userName,
          departmentId: deparmentList.indexOf(departmentId) + 1,
        }
      );
      const token = response.data.token;
      const decodedToken = parseJwt(token);
      const userName = decodedToken.sub;
      localStorage.setItem("token", token);
      navigate("/mainNavigation", { state: { userName } });
    } catch (error) {
      console.error("Login failed", error);
    }
  };

  const parseJwt = (token) => {
    const base64Url = token.split(".")[1];
    const base64 = base64Url.replace(/-/g, "+").replace(/_/g, "/");
    const jsonPayload = decodeURIComponent(
      atob(base64)
        .split("")
        .map((c) => "%" + ("00" + c.charCodeAt(0).toString(16)).slice(-2))
        .join("")
    );
    return JSON.parse(jsonPayload);
  };

  return (
    <div className="student-signup-main-container">
      <div
        className="closeCreateEventWindow"
        onClick={handlecloseCreateStudentWindow}
      >
        <IoMdCloseCircleOutline />
      </div>
      <p className="student-head1">Create Student</p>
      <div className="student-login-container">
        <div className="student-image-container">
          <img src={Designer} className="student-logo" alt="Logo" />
        </div>
        <div className="form-container">
          <form onSubmit={handleSubmit}>
            {/* First Name Input */}
            <InputField
              placeholder="Enter your first name"
              value={firstname}
              onChange={(e) => setFirstName(e.target.value)}
              inputType="text"
            />

            {/* Last Name Input */}
            <InputField
              placeholder="Enter your last name"
              value={lastname}
              onChange={(e) => setLastName(e.target.value)}
              inputType="text"
            />

            {/* Email Input */}
            <InputField
              placeholder="Enter your email"
              value={email}
              onChange={(e) => setEmail(e.target.value)}
              inputType="text"
            />

            <InputField
              placeholder="Enter your user name"
              value={userName}
              onChange={(e) => setUserName(e.target.value)}
              inputType="text"
            />

            {/* Password Input */}
            <InputField
              placeholder="Enter your password"
              value={password}
              onChange={(e) => setPassword(e.target.value)}
              inputType="password"
            />

            <div className="choice-input mb-3">
              <select
                value={departmentId}
                onChange={(e) => setDepartmentId(e.target.value)}
                className="student-select-input"
              >
                <option value="">Select the department</option>
                {deparmentList.map((option, index) => (
                  <option key={index} value={option}>
                    {option}
                  </option>
                ))}
              </select>
            </div>
            <button type="submit" className="btn btn-primary w-100">
              Add Student
            </button>
          </form>
        </div>
      </div>
    </div>
  );
};

export default CreateStudentWindow;
