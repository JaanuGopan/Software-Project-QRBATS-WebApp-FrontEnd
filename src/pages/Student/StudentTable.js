import React, { useState, useEffect } from "react";
import "../AdminDashboard/AdminDashboard.css";
import { FaEdit } from "react-icons/fa";

const StudentTable = ({ handleUpdateStudent, onStudentClick, studentList }) => {
  const [selectedStudent, setSelectedStudent] = useState(null);

  const handleStudentClick = (student) => {
    onStudentClick(student);
    setSelectedStudent(student);
  };

  const departmentList = ["DEIE", "DCOM", "DMME", "DCEE", "DMENA", "DIS"];

  return (
    <div className="tableDesign">
      <table className="student-tableArrangement">
        <thead>
          <tr>
            <th>No</th>
            <th className="expand">Name</th>
            <th>IndexNo</th>
            <th>Department</th>
            <th>Semester</th>
            <th>Edit</th>
          </tr>
        </thead>
        <tbody>
          {studentList.map((student, index) => (
            <tr
              key={index}
              onClick={() => handleStudentClick(student)}
              className={
                selectedStudent === student ? "selected-row" : "event-row"
              }
            >
              <td>{index + 1}</td>
              <td>{student.studentName}</td>
              <td>{student.indexNumber}</td>
              <td>{departmentList[student.departmentId - 1]}</td>
              <td>{student.currentSemester}</td>
              <td>
                <button
                  onClick={() => handleUpdateStudent(student)}
                  className="EditButton"
                >
                  <FaEdit className="EditIcon" />
                </button>
              </td>
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  );
};

export default StudentTable;
