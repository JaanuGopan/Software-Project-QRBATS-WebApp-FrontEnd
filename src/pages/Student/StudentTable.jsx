import React, { useState, useEffect } from 'react';
import '../AdminDashboard/AdminDashboard.css';
import { FaEdit } from 'react-icons/fa';

const StudentTable = ({ handleUpdateStudent, onStudentClick, studentList, search }) => {
  const [selectedStudent, setSelectedStudent] = useState(null);
  
  const handleStudentClick = (student) => {
    onStudentClick(student);
    setSelectedStudent(student);
  };

  const departmentList = ['DEIE', 'DCOM', 'DMME', 'DCEE', 'DMENA', 'DIS'];

  return (
    <div className="tableDesign">
      <table className="student-tableArrangement">
        <thead>
          <tr>
            <th>No</th>
            <th className="expand">Name</th>
            <th>IndexNo</th>
            <th>Department</th>
            <th>Semester</th>
            <th>Edit</th>
          </tr>
        </thead>
        {studentList.filter(
              (student) => student.firstName.toLowerCase().includes(search.toLowerCase()) ||
              student.indexNumber.toLowerCase().includes(search.toLowerCase())
            ).length > 0 ? <tbody>
          {studentList.filter(
              (student) => student.firstName.toLowerCase().includes(search.toLowerCase()) ||
              student.indexNumber.toLowerCase().includes(search.toLowerCase())
            )
            .map((student, index) => (
            <tr
              key={index}
              onClick={() => handleStudentClick(student)}
              className={selectedStudent === student ? 'selected-row' : 'event-row'}
            >
              <td>{index + 1}</td>
              <td>{student.firstName}</td>
              <td>{student.indexNumber}</td>
              <td>{departmentList[student.departmentId - 1]}</td>
              <td>{student.semester}</td>
              <td>
                <button onClick={() => handleUpdateStudent(student)} className="EditButton">
                  <FaEdit className="EditIcon" />
                </button>
              </td>
            </tr>
          ))}
        </tbody> : <tbody className='nodata'>
          No Data Available
        </tbody>}
      </table>
    </div>
  );
};

export default StudentTable;
