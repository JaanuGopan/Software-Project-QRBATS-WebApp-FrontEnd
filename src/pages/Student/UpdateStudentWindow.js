import React, { useState } from "react";
import "./Student.css";
import { useNavigate } from "react-router-dom";
import InputField from "../../components/textfields/InputBox/InputField";
import { IoMdCloseCircleOutline } from "react-icons/io";
import Select from "react-select";
import StudentService from "../../api/services/StudentService";
import { ToastContainer, toast } from "react-toastify";

const UpdateStudentWindow = ({
  handleCloseUpdateStudentWindow,
  student,
  handleReloadStudentList,
}) => {
  const [studentId, setStudentId] = useState(student.studentId);
  const [indexNumber, setIndexNumber] = useState(student.indexNumber);
  const [studentName, setStudentName] = useState(student.studentName);
  const [studentEmail, setStudentEmail] = useState(student.studentEmail);
  const [password, setPassword] = useState(student.password);
  const [userName, setUserName] = useState(student.username);
  const [StudentRole, setStudentRole] = useState(student.studentRole);
  const [departmentId, setDepartmentId] = useState(student.departmentId);
  const departmentList = ["DEIE", "DCOM", "DMME", "DCEE", "DMENA", "DIS"];

  const [department, setDepartment] = useState({
    value: departmentList[student.departmentId - 1],
    label: departmentList[student.departmentId - 1],
  });
  const semesterList = ["1", "2", "3", "4", "5", "6", "7", "8"];
  const [semester, setSemester] = useState({
    value: semesterList[student.currentSemester - 1],
    label: semesterList[student.currentSemester - 1],
  });

  const [updateSuccess, setUpdateSuccess] = useState(false);

  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      const response = await StudentService.updateStudent(
        studentId,
        studentName,
        indexNumber,
        studentEmail,
        userName,
        departmentList.indexOf(department.value) + 1,
        semesterList.indexOf(semester.value) + 1
      );
      if (response.status === 200) {
        toast.success("Successfully Updated.");
        setUpdateSuccess(true);
      } else {
        toast.error(response);
      }
    } catch (error) {
      console.error("Login failed", error);
    } finally {
      handleReloadStudentList();
      if (updateSuccess) {
        handleCloseUpdateStudentWindow();
      }
    }
  };

  return (
    <div className="student-signup-main-container">
      <ToastContainer />
      <div className="student-create-title-close-button">
        <h3 className="student-create-title">Update Student</h3>
        <div
          className="student-create-close-button"
          onClick={handleCloseUpdateStudentWindow}
        >
          <IoMdCloseCircleOutline id="close-icon" />
        </div>
      </div>
      <div className="student-login-container">
        <div className="form-container">
          <form onSubmit={handleSubmit}>
            <div className="student-creation-input">
              <label>Student Name</label>
              <div className="student-creation-input-field">
                <InputField
                  placeholder="Enter Student name"
                  value={studentName}
                  onChange={(e) => setStudentName(e.target.value)}
                  inputType="text"
                />
              </div>
            </div>
            <div className="student-creation-input">
              <label>IndexNumber</label>
              <div className="student-creation-input-field">
                <InputField
                  placeholder="Enter Student IndexNo"
                  value={indexNumber}
                  onChange={(e) => setIndexNumber(e.target.value)}
                  inputType="text"
                />
              </div>
            </div>
            <div className="student-creation-input">
              <label>Student Email</label>
              <div className="student-creation-input-field">
                <InputField
                  placeholder="Enter Student Email"
                  value={studentEmail}
                  onChange={(e) => setStudentEmail(e.target.value)}
                  inputType="text"
                />
              </div>
            </div>
            <div className="student-creation-input">
              <label>Semester</label>
              <div className="student-creation-input-field">
                <Select
                  required
                  placeholder="Enter Department"
                  value={semester}
                  onChange={(e) => setSemester(e)}
                  options={semesterList.map((sem) => ({
                    value: sem,
                    label: sem,
                  }))}
                />
              </div>
            </div>

            <div className="student-creation-input">
              <label>Department</label>
              <div className="student-creation-input-field">
                <Select
                  required
                  placeholder="Enter Department"
                  value={department}
                  onChange={(e) => setDepartment(e)}
                  options={departmentList.map((department) => ({
                    value: department,
                    label: department,
                  }))}
                />
              </div>
            </div>
            {/* <div className="student-creation-input">
              <label>UserName</label>
              <div className="student-creation-input-field">
                <InputField
                  placeholder="Enter Student UserName"
                  value={userName}
                  onChange={(e) => setUserName(e.target.value)}
                  inputType="text"
                />
              </div>
            </div> */}

            <div className="update-student-buttons-container">
              <button
                type="submit"
                className="btn btn-warning update-student-save-button"
              >
                Save
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  );
};

export default UpdateStudentWindow;
