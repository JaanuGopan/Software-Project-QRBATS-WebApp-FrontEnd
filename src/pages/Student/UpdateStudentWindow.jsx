import React, { useState } from 'react';
import './Student.css';
import InputField from '../../components/textfields/InputBox/InputField';
import { IoMdCloseCircleOutline } from 'react-icons/io';
import Select from 'react-select';
import StudentService from '../../api/services/StudentService';
import { toast } from 'react-toastify';
import { CircularProgress } from '@mui/material';
import Department from '../../utils/Department';

const UpdateStudentWindow = ({
  handleCloseUpdateStudentWindow,
  student,
  handleReloadStudentList,
}) => {
  const [studentId, setStudentId] = useState(student.userId);
  const [indexNumber, setIndexNumber] = useState(student.indexNumber);
  const [studentName, setStudentName] = useState(student.firstName);
  const [studentEmail, setStudentEmail] = useState(student.email);
  const [userName, setUserName] = useState(student.username);
  const [StudentRole, setStudentRole] = useState(student.role);
  const [departmentId, setDepartmentId] = useState(student.departmentId);
  const departmentList = Department.studentDepartmentList;

  const [department, setDepartment] = useState({
    value: departmentList[student.departmentId - 1],
    label: departmentList[student.departmentId - 1],
  });
  const semesterList = ['1', '2', '3', '4', '5', '6', '7', '8'];
  const [semester, setSemester] = useState({
    value: semesterList[student.semester - 1],
    label: semesterList[student.semester - 1],
  });

  const [updateSuccess, setUpdateSuccess] = useState(false);

  const [processing, setProcessing] = useState(false);

  const handleInputValidation = () => {
    const emailPattern = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;

    if (!studentName.trim()) {
      toast.error('Student name is required');
      return false;
    }
    // indexno = "EG/20XX/XXXX"
    const indexNoPattern = /^EG\/20[0-9]{2}\/[0-9]{4}$/;

    if (!indexNoPattern.test(indexNumber)) {
      toast.error('Invalid index number');
      return false;
    }

    if (!indexNumber.trim()) {
      toast.error('Index number is required');
      return false;
    }

    if (!studentEmail.trim() || !emailPattern.test(studentEmail)) {
      toast.error('A valid email is required');
      return false;
    }

    if (!userName.trim()) {
      toast.error('Username is required');
      return false;
    }

    if (!department.value) {
      toast.error('Department is required');
      return false;
    }

    if (!semester.value) {
      toast.error('Semester is required');
      return false;
    }

    return true;
  };

  const handleSubmit = async (e) => {
    e.preventDefault();
    if (!handleInputValidation()) {
      return;
    }

    try {
      setProcessing(true);
      const response = await StudentService.updateStudent(
        studentId,
        studentName,
        indexNumber,
        studentEmail,
        userName,
        departmentList.indexOf(department.value) + 1,
        semesterList.indexOf(semester.value) + 1,
        StudentRole
      );
      if (response.status === 200) {
        toast.success('Successfully Updated.');
        setUpdateSuccess(true);
        handleCloseUpdateStudentWindow();
        handleReloadStudentList();
      } else if (response.status === 400) {
        toast.error(response.data);
      }
    } catch (error) {
      console.error('Update failed', error);
      toast.error('Error In Updating Student.');
    } finally {
      setProcessing(false);
    }
  };

  return (
    <div className="student-signup-main-container">
      <div className="student-create-title-close-button">
        <h3 className="student-create-title">Update Student</h3>
        <div className="student-create-close-button" onClick={handleCloseUpdateStudentWindow}>
          <IoMdCloseCircleOutline id="close-icon" />
        </div>
      </div>
      <div className="student-login-container">
        <div className="form-container">
          <form onSubmit={handleSubmit}>
            <div className="student-creation-input">
              <label>Student Name</label>
              <div className="student-creation-input-field">
                <InputField
                  placeholder="Enter Student name"
                  value={studentName}
                  onChange={(e) => setStudentName(e.target.value)}
                  inputType="text"
                />
              </div>
            </div>
            <div className="student-creation-input">
              <label>Index Number</label>
              <div className="student-creation-input-field">
                <InputField
                  placeholder="Enter Student Index No"
                  value={indexNumber}
                  onChange={(e) => setIndexNumber(e.target.value)}
                  inputType="text"
                />
              </div>
            </div>
            <div className="student-creation-input">
              <label>Student Email</label>
              <div className="student-creation-input-field">
                <InputField
                  placeholder="Enter Student Email"
                  value={studentEmail}
                  onChange={(e) => setStudentEmail(e.target.value)}
                  inputType="text"
                />
              </div>
            </div>
            <div className="student-creation-input">
              <label>Semester</label>
              <div className="student-creation-input-field">
                <Select
                  required
                  placeholder="Enter Semester"
                  value={semester}
                  onChange={(e) => setSemester(e)}
                  options={semesterList.map((sem) => ({
                    value: sem,
                    label: sem,
                  }))}
                />
              </div>
            </div>
            <div className="student-creation-input">
              <label>Department</label>
              <div className="student-creation-input-field">
                <Select
                  required
                  placeholder="Enter Department"
                  value={department}
                  onChange={(e) => setDepartment(e)}
                  options={departmentList.map((department) => ({
                    value: department,
                    label: department,
                  }))}
                />
              </div>
            </div>
            {/* <div className="student-creation-input">
              <label>UserName</label>
              <div className="student-creation-input-field">
                <InputField
                  placeholder="Enter Student UserName"
                  value={userName}
                  onChange={(e) => setUserName(e.target.value)}
                  inputType="text"
                />
              </div>
            </div> */}

            <div className="update-student-buttons-container">
              {processing ? (
                <CircularProgress />
              ) : (
                <button type="submit" className="btn btn-warning update-student-save-button">
                  Save
                </button>
              )}
            </div>
          </form>
        </div>
      </div>
    </div>
  );
};

export default UpdateStudentWindow;
