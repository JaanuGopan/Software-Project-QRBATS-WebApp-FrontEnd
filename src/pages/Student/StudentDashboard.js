import React, { useState } from "react";
import "./Student.css";
import NormalButton from "../../components/layout/AdminDashboardComponent/NormalButton";
import { MdCreateNewFolder } from "react-icons/md";
import { RiDeleteBin5Fill } from "react-icons/ri";
import StudentTable from "../../components/layout/AdminDashboardComponent/StudentTable";
import CreateStudentWindow from "./CreateStudentWindow";
import UpdateStudentWindow from "./UpdateStudentWindow";

const StudentDashboard = () => {
  const [studentCreatePopUpWindow, setStudentCreatePopUpWindow] =
    useState(false);
  const [studentUpdatePopUpWindow, setStudentUpdatePopUpWindow] =
    useState(false);

  return (
    <div className="student-Dash">
      <div className="student-SearchEvent">
        <p className="student-mainHead">Student Details</p>
        <input
          type="text"
          placeholder="Search..."
          style={{
            width: "150px",
            padding: "3px 40px",
            border: "0.5px solid black",
            borderRadius: "5px",
            textAlign: "center",
          }}
        />
      </div>
      <div className="student-EventList">
        <StudentTable
          handleUpdateStudent={() => setStudentUpdatePopUpWindow(true)}
        />
        <div className="student-List-Buttons">
          <NormalButton
            handleClick={() => setStudentCreatePopUpWindow(true)}
            title={"Create"}
            titlewithiconicon={
              <MdCreateNewFolder className="student-buttonIcon" />
            }
          />
          <NormalButton
            title={"Delete"}
            titlewithiconicon={
              <RiDeleteBin5Fill className="student-buttonIcon" />
            }
          />
        </div>
      </div>
      {studentCreatePopUpWindow && (
        <div
          handleClick={() => setStudentCreatePopUpWindow(false)}
          className="student-Create-Event-Dashboard"
        >
          <CreateStudentWindow
            handlecloseCreateStudentWindow={() =>
              setStudentCreatePopUpWindow(false)
            }
          />
        </div>
      )}
      {studentUpdatePopUpWindow && (
        <div
          handleClick={() => setStudentCreatePopUpWindow(false)}
          className="student-Create-Event-Dashboard"
        >
          <UpdateStudentWindow
            handlecloseUpdateStudentWindow={() =>
              setStudentUpdatePopUpWindow(false)
            }
          />
        </div>
      )}
    </div>
  );
};

export default StudentDashboard;
