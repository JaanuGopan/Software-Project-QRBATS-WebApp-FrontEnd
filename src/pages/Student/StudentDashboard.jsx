import React, { useEffect, useState } from 'react';
import './Student.css';
import NormalButton from '../../components/layout/AdminDashboardComponent/NormalButton';
import { MdCreateNewFolder } from 'react-icons/md';
import { RiDeleteBin5Fill } from 'react-icons/ri';
import StudentTable from './StudentTable';
import CreateStudentWindow from './CreateStudentWindow';
import UpdateStudentWindow from './UpdateStudentWindow';
import StudentService from '../../api/services/StudentService';
import WarningPopup from '../../components/warningPopup/WarningPopup';
import { ToastContainer, toast } from 'react-toastify';

const StudentDashboard = () => {
  const [studentCreatePopUpWindow, setStudentCreatePopUpWindow] = useState(false);
  const [studentUpdatePopUpWindow, setStudentUpdatePopUpWindow] = useState(false);
  const [showDeletePopUpWindow, setShowDeletePopUpWindow] = useState(false);
  const [selectedStudent, setSelectedStudent] = useState(null);
  const [studentList, setStudentList] = useState([]);
  const [search, setSearch] = useState('');

  const handleSelectedStudent = (student) => {
    setSelectedStudent(student);
  };

  const handleReloadStudentList = async () => {
    const response = await StudentService.getAllStudent();
    if (response) {
      setStudentList(response);
    }
  };

  useEffect(() => {
    handleReloadStudentList();
  }, []);

  const [processingDeleteStudent, setProcessingDeleteStudent] = useState(false);
  const handelDeleteStudent = async () => {
    try {
      setProcessingDeleteStudent(true);
      const response = await StudentService.deleteStudent(selectedStudent.userId);
      if (response) {
        handleReloadStudentList();
        setShowDeletePopUpWindow(false);
        toast.success('Successfully Deleted.');
        setSelectedStudent(null);
      }
    } finally {
      setProcessingDeleteStudent(false);
    }
  };

  return (
    <div className="student-Dash">
      <ToastContainer />
      <div className="student-SearchEvent">
        <p className="student-mainHead">Student Details</p>
        <input
          type="text"
          placeholder="Search..."
          style={{
            width: '150px',
            padding: '3px 40px',
            border: '0.5px solid black',
            borderRadius: '5px',
            textAlign: 'center',
          }}
          onChange={(e) => setSearch(e.target.value)}
        />
      </div>
      <div className="student-EventList">
        <StudentTable
          handleUpdateStudent={(e) => {
            setSelectedStudent(e);
            setStudentUpdatePopUpWindow(true);
          }}
          onStudentClick={(e) => handleSelectedStudent(e)}
          studentList={studentList}
          search={search}
        />
        <div className="student-List-Buttons">
          <NormalButton
            handleClick={() => setStudentCreatePopUpWindow(true)}
            title={'Create'}
            titlewithiconicon={<MdCreateNewFolder className="student-buttonIcon" />}
          />
          <NormalButton
            title={'Delete'}
            titlewithiconicon={<RiDeleteBin5Fill className="student-buttonIcon" />}
            handleClick={() => {
              if (!selectedStudent) {
                toast.error('Please select a student to delete.');
              } else {
                setShowDeletePopUpWindow(true);
              }
            }}
          />
        </div>
      </div>
      {studentCreatePopUpWindow && (
        <div className="student-Create-Event-Dashboard">
          <CreateStudentWindow
            handleCloseCreateStudentWindow={() => setStudentCreatePopUpWindow(false)}
            handleReloadStudentList={handleReloadStudentList}
          />
        </div>
      )}
      {studentUpdatePopUpWindow && (
        <div className="student-Create-Event-Dashboard">
          <UpdateStudentWindow
            handleCloseUpdateStudentWindow={() => setStudentUpdatePopUpWindow(false)}
            student={selectedStudent}
            handleReloadStudentList={handleReloadStudentList}
          />
        </div>
      )}
      {showDeletePopUpWindow && selectedStudent && (
        <div className="student-creation-delete-popup-window">
          <WarningPopup
            handleOk={handelDeleteStudent}
            handleCloseWarningWindow={() => setShowDeletePopUpWindow(false)}
            buttonText={'Delete'}
            titleText={'Are you sure you want to delete this student?'}
            processing={processingDeleteStudent}
          />
        </div>
      )}
    </div>
  );
};

export default StudentDashboard;
