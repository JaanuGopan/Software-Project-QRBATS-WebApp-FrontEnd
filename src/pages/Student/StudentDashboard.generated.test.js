import renderer from 'react-test-renderer';
import React, { useState } from "react";

import NormalButton from "../../components/layout/AdminDashboardComponent/NormalButton";
import { MdCreateNewFolder } from "react-icons/md";
import { RiDeleteBin5Fill } from "react-icons/ri";
import StudentTable from "../../components/layout/AdminDashboardComponent/StudentTable";
import CreateStudentWindow from "./CreateStudentWindow";
import UpdateStudentWindow from "./UpdateStudentWindow";
import StudentDashboard from "./StudentDashboard";

jest.mock("./Student.css");
jest.mock("../../components/layout/AdminDashboardComponent/NormalButton");
jest.mock("react-icons/md");
jest.mock("react-icons/ri");
jest.mock("../../components/layout/AdminDashboardComponent/StudentTable");
jest.mock("./CreateStudentWindow");
jest.mock("./UpdateStudentWindow");

const renderTree = tree => renderer.create(tree);
describe('<StudentDashboard>', () => {
  it('should render component', () => {
    expect(renderTree(<StudentDashboard 
    />).toJSON()).toMatchSnapshot();
  });
  
});