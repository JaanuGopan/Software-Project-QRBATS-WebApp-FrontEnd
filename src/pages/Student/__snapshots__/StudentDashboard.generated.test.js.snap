// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StudentDashboard> should render component 1`] = `
<div
  className="student-Dash"
>
  <div
    className="student-SearchEvent"
  >
    <p
      className="student-mainHead"
    >
      Student Details
    </p>
    <input
      placeholder="Search..."
      style={
        Object {
          "border": "0.5px solid black",
          "borderRadius": "5px",
          "padding": "3px 40px",
          "textAlign": "center",
          "width": "150px",
        }
      }
      type="text"
    />
  </div>
  <div
    className="student-EventList"
  >
    <div
      className="student-List-Buttons"
    />
  </div>
</div>
`;
